enterprise-suite-latest/Makefile

```diff
@@ -0,0 +1,40 @@
+CHART = enterprise-suite-latest
+
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!
+#
+# This is a very atypical chart Makefile.  (aka hack)  It's really just here so we can build
+# a 'latest' version of enterprise-suite.
+#
+# Do not use this as a template for a new chart project.
+#
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+BASE_CHART = enterprise-suite
+HELM_CHARTS_DIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
+SCRIPTS_DIR := $(HELM_CHARTS_DIR)/scripts
+
+# Defines before include to tweak behavior as required
+VERSION = $(shell $(SCRIPTS_DIR)/export-chart-version.sh $(BASE_CHART))
+ALL_YAML = all-latest.yaml
```

Comment:
I'm not sure about this - it's trying to pass arguments in via global variables defined in another file. Feels pretty hacky and easy to break or misunderstand. Can we do this another way - `common.mk` should be as simple as possible without much conditional logic. Even if it means copy pasting some of common.mk in here, I think that's better.

Reply:
It is hacky (and that's acknowledged in the comments).  What we're trying to do with `enterprise-suite` and `enterprise-suite-latest` is a hack and the requisite ugliness has to show up _somewhere_.  The current design limits the ugliness to `enterprise-suite-latest/Makefile` to the extent that it can.  (There is some very small leakage into `common.mk` but not in a way that a client would care/notice.)  (See for e.g. `reactive-sandbox/Makefile` or `sample-project/Makefile`.)

An alternative design would be to symlink everything from `enterprise-suite` into `enterprise-suite-latest` and then `enterprise-suite-latest` could be treated like "any other helm-charts client."  That felt _very_ ugly to me though (with confusing 'duplication', and the way Git handles symlinks).

If we went the copy-paste route, then I suspect we'd pretty much duplicate `common.mk` and then we'd risk fixing things in one place and not the other.  I'd argue strongly against going that route.

Reply:
As long as this file is only required for ES charts, that is okay. My goal is to avoid having common dependencies across team boundaries, which makes that code very difficult to change. Also, imo, duplication in make is usually preferable to make madness (automake, config tools, macros, etc.).